core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -307,6 +331,71 @@ trait WorkspaceComponent {
       }
     }
 
+    def insertUserCatalogPermissions(workspaceId: UUID, userRefs: Seq[RawlsUserRef]) = {
+      val recordsToInsert = userRefs.map(userRef => WorkspaceUserCatalogRecord(workspaceId, userRef.userSubjectId.value))
+      (workspaceUserCatalogQuery ++= recordsToInsert)
+    }
+
+    def insertGroupCatalogPermissions(workspaceId: UUID, groupRefs: Seq[RawlsGroupRef]) = {
+      val recordsToInsert = groupRefs.map(groupRef => WorkspaceGroupCatalogRecord(workspaceId, groupRef.groupName.value))
+      (workspaceGroupCatalogQuery ++= recordsToInsert)
+    }
+
+    def deleteUserCatalogPermissions(workspaceId: UUID, userRefs: Seq[RawlsUserRef]) = {
+      val usersToRemove = userRefs.map(_.userSubjectId.value)
+      (workspaceUserCatalogQuery.filter(rec => rec.workspaceId === workspaceId && rec.userSubjectId.inSet(usersToRemove))).delete
+    }
+
+    def deleteGroupCatalogPermissions(workspaceId: UUID, groupRefs: Seq[RawlsGroupRef]) = {
+      val groupsToRemove = groupRefs.map(_.groupName.value)
+      (workspaceGroupCatalogQuery.filter(rec => rec.workspaceId === workspaceId && rec.groupName.inSet(groupsToRemove))).delete
+    }
+
+    def deleteWorkspaceCatalogPermissions(workspaceId: UUID) = {
```

Comment:
this should be used in `deleteWorkspace` in WorkspaceService.scala like how `deleteWorkspaceSharePermissions` is used, otherwise we'll run into foreign key errors when trying to delete workspaces that have these permissions set